server/channels/app/support_packet.go

```diff
@@ -103,140 +124,129 @@ func (a *App) generateSupportPacketYaml() (*model.FileData, string) {
+		supportPacket.InactiveUserCount = int(analytics[10].Value)
 	}
 
 	// Marshal to a Yaml File
 	supportPacketYaml, err := yaml.Marshal(&supportPacket)
-	if err == nil {
-		fileData := model.FileData{
-			Filename: "support_packet.yaml",
-			Body:     supportPacketYaml,
-		}
-		return &fileData, ""
+	if err != nil {
+		rErr = multierror.Append(errors.Wrap(err, "failed to marshal support package into yaml"))
 	}
 
-	warning := fmt.Sprintf("yaml.Marshal(&supportPacket) Error: %s", err.Error())
-	return nil, warning
+	fileData := &model.FileData{
+		Filename: "support_packet.yaml",
+		Body:     supportPacketYaml,
+	}
+	return fileData, rErr
 }
 
-func (a *App) createPluginsFile() (*model.FileData, string) {
-	var warning string
-
+func (a *App) createPluginsFile() (*model.FileData, error) {
 	// Getting the plugins installed on the server, prettify it, and then add them to the file data array
 	pluginsResponse, appErr := a.GetPlugins()
-	if appErr == nil {
-		pluginsPrettyJSON, err := json.MarshalIndent(pluginsResponse, "", "    ")
-		if err == nil {
-			fileData := model.FileData{
-				Filename: "plugins.json",
-				Body:     pluginsPrettyJSON,
-			}
-
-			return &fileData, ""
-		}
-
-		warning = fmt.Sprintf("json.MarshalIndent(pluginsResponse) Error: %s", err.Error())
-	} else {
-		warning = fmt.Sprintf("c.App.GetPlugins() Error: %s", appErr.Error())
+	if appErr != nil {
+		return nil, errors.Wrap(appErr, "failed to get plugin list for support package")
 	}
 
-	return nil, warning
-}
-
-func (a *App) getNotificationsLog() (*model.FileData, string) {
-	var warning string
-
-	// Getting notifications.log
-	if *a.Config().NotificationLogSettings.EnableFile {
-		// notifications.log
-		notificationsLog := config.GetNotificationsLogFileLocation(*a.Config().LogSettings.FileLocation)
-
-		notificationsLogFileData, notificationsLogFileDataErr := os.ReadFile(notificationsLog)
-
-		if notificationsLogFileDataErr == nil {
-			fileData := model.FileData{
-				Filename: "notifications.log",
-				Body:     notificationsLogFileData,
-			}
-			return &fileData, ""
-		}
-
-		warning = fmt.Sprintf("os.ReadFile(notificationsLog) Error: %s", notificationsLogFileDataErr.Error())
+	pluginsPrettyJSON, err := json.MarshalIndent(pluginsResponse, "", "    ")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal plugin list into json")
+	}
 
-	} else {
-		warning = "Unable to retrieve notifications.log because LogSettings: EnableFile is false in config.json"
+	fileData := &model.FileData{
+		Filename: "plugins.json",
+		Body:     pluginsPrettyJSON,
 	}
+	return fileData, nil
 
-	return nil, warning
 }
 
-func (a *App) getMattermostLog() (*model.FileData, string) {
-	var warning string
+func (a *App) getNotificationsLog() (*model.FileData, error) {
+	if !*a.Config().NotificationLogSettings.EnableFile {
+		return nil, errors.New("Unable to retrieve notifications.log because LogSettings: EnableFile is false in config.json")
```

Comment:
It's trivial but config doesn't always needs to be in `config.json`. We also have a config in the DB.

Reply:
See [3d4a97f](https://github.com/mattermost/mattermost/pull/24382/commits/3d4a97f831143cf8414678768d4700c052cc799f)